documentation/modules/oauth/con-oauth-authentication-broker.adoc

```diff
@@ -0,0 +1,138 @@
+// Module included in the following assemblies:
+//
+// assembly-oauth.adoc
+
+[id='con-oauth-authentication-broker-{context}']
+= {oauth} Kafka broker configuration
+
+Kafka broker configuration for {oauth} involves:
+
+* Creating the {oauth} client in the authorization server
+* Configuring {oauth} authentication in the Kafka custom resource
+
+NOTE: In relation to the authorization server, Kafka brokers and Kafka clients are both regarded as {oauth} clients.
+
+== {oauth} client configuration on an authorization server
+
+To configure a Kafka broker to validate the token received during session initiation,
+the recommended approach is to create an {oauth} _client_ definition in an authorization server, configured as _confidential_, with client credentials enabled:
+
+* Client ID of `Kafka`
+* Client ID and Secret as the authentication mechanism
+
+NOTE: You only need to use a client ID and secret when using a non-public introspection endpoint of the authorization server.
+The credentials are not typically required when using public authorization server endpoints, as with fast local JWT token validation.
+
+== {oauth} authentication configuration in the Kafka cluster
+
+To use {oauth} authentication in the Kafka cluster, you specify a TLS listener configuration for your Kafka cluster custom resource with the authentication method `oauth`:
+
+.Authentication method type for {oauth}
+[source,yaml,subs="+quotes, attributes"]
+----
+apiVersion: {KafkaApiVersion}
+kind: Kafka
+spec:
+  kafka:
+    listeners:
+      *tls*:
+        authentication:
+          type: *oauth*
+          #...
+----
+
+You can configure `plain`, `tls` and `external` listeners, as described in xref:assembly-configuring-kafka-broker-listeners-deployment-configuration-kafka[Kafka broker listeners],
+but it is recommended not to use `plain` listeners with {oauth} as this creates a vulnerability to network eavesdropping and unauthorized access through token theft.
+
+You configure an `external` listener with `tls: true` for a secure transport layer to communicate with the client.
+
+.External listener with TLS
+[source,yaml,subs="+quotes"]
+----
+# ...
+listeners:
+  tls:
+    authentication:
+      type: oauth
+  external:
+    type: oauth
+    *tls: true*
+    authentication:
+      type: tls
+# ...      #...
```

Comment:
Same as above ... Should this have some complete example? Users will be copy pasting and will be surprised it does not work.

Reply:
I'm not sure we need to repeat what we have in the procedures. But to emphasize that it's not the end of the story, I've added:

> When you've defined the type of authentication as {oauth}, you add configuration based on the type of validation, either as <<con-oauth-authentication-broker-fast-local, fast local JWT validation>> or <<con-oauth-authentication-broker-intro-local, token validation using an introspection endpoint>>.
> 
> The procedure to configure {oauth} for listeners, with descriptions and examples, is described in xref:proc-oauth-broker-config-{context}[Configuring {oauth} support for Kafka brokers].
>

Reply:
Right. The problem is that the reader doesn't know there will be a procedure later. He will think that this is it. The added test makes it clear so that works for me.